src/number_theory/number_field.lean

```diff
@@ -142,3 +146,179 @@ noncomputable def ring_of_integers_equiv : ring_of_integers ℚ ≃+* ℤ :=
 ring_of_integers.equiv ℤ
 
 end rat
+
+namespace adjoin_root
+
+section
+
+open_locale polynomial
+
+local attribute [-instance] algebra_rat
+
+/-- The quotient of `ℚ[X]` by the ideal generated by an irreducible polynomial of `ℚ[X]`
+is a number field. -/
+instance {f : ℚ[X]} [hf : irreducible f] : number_field (adjoin_root f) :=
+{ to_char_zero := char_zero_of_injective_algebra_map (algebra_map ℚ _).injective,
+  to_finite_dimensional := begin
+   let := (adjoin_root.power_basis (irreducible.ne_zero hf : f ≠ 0)),
+   convert power_basis.finite_dimensional this,
+   haveI : subsingleton (algebra ℚ (adjoin_root f)) := algebra_rat_subsingleton,
+   exact subsingleton.elim _ _,
+  end }
+
+end
+
+end adjoin_root
+
+namespace number_field.embeddings
+
+open set finite_dimensional complex
+open_locale complex_conjugate
+
+variables {K L : Type*} [field K] [field L]
+section number_field
+variables [number_field K] [number_field L]  (x : K)
+
+/-- The equivalence between algebra maps from a number field to `ℂ` and plain
+ring morphisms between them. -/
+noncomputable def equiv_alg : (K →ₐ[ℚ] ℂ) ≃ (K →+* ℂ) :=
+{ to_fun := coe,
+  inv_fun := λ f : K →+* ℂ, alg_hom.mk' f (λ (c : ℚ) x, map_rat_smul f _ _),
+  left_inv  := λ x, alg_hom.ext  $ by simp only [forall_const, alg_hom.coe_to_ring_hom,
+                                                 eq_self_iff_true, alg_hom.coe_mk'],
+  right_inv := λ x, ring_hom.ext $ by simp only [forall_const, alg_hom.coe_to_ring_hom,
+                                                 eq_self_iff_true, alg_hom.coe_mk'] }
+
+/-- There are finitely many embeddings of a number field. -/
+noncomputable instance : fintype (K →+* ℂ) := fintype.of_equiv (K →ₐ[ℚ] ℂ) equiv_alg
+
+/-- The number of embeddings of a number field is its finrank. -/
+lemma card_embeddings : fintype.card (K →+* ℂ) = finrank ℚ K :=
+by rw [fintype.of_equiv_card equiv_alg, alg_hom.card]
+
+open polynomial
+
+/-- Any embeddings of a number field `K` can be extended to an embedding of an extension `L`. -/
+lemma lift {L : Type*} [field L] [number_field L] [algebra K L] (φ : K →+* ℂ) :
```

Comment:
This is literally [is_alg_closed.lift](https://leanprover-community.github.io/mathlib_docs/field_theory/is_alg_closed/basic.html#is_alg_closed.lift) (that you use in the proof), I don't understand why you need to restate it.

Reply:
I agree. It's gone now.